packages/core/test/types.test.ts

```diff
@@ -480,3 +493,90 @@ describe('spawn', () => {
     });
   });
 });
+
+describe('createMachine', () => {
+  it('should type implementations correctly', () => {
+    createMachine2<{
+      events: { type: 'foo' };
+      actions: { type: 'greet' };
+      guards: { type: 'isValid'; params: {} }; // TODO: make params optional
+      actors:
+        | {
+            src: 'notifier';
+            data: string | undefined;
+          }
+        | {
+            src: 'notifier2';
+            data: string | undefined;
+          };
+      delays: {
+        timeout: number;
+      };
+      input: {
+        message: string;
+      };
+    }>(
+      {},
+      {
+        actions: {
+          greet: () => {},
+          // @ts-expect-error
+          nonexistant: () => {}
+        },
+        delays: {
+          timeout: 234,
+          // @ts-expect-error
+          other: 123
+        },
+        guards: {
+          isValid: () => true,
+          // @ts-expect-error
+          nonexistant: () => false
+        },
+        actors: {
+          notifier: () => fromPromise(() => Promise.resolve('test')),
+          // @ts-expect-error
+          notifier2: () => fromPromise(() => Promise.resolve(42)),
+          // @ TODO: this should be an error
+          nonexistant: () => fromPromise(() => Promise.resolve(42))
+        },
+        input: {
+          message: 'asdf',
+          // @ts-expect-error
+          other: 42
+        }
+      }
+    );
+  });
+});
+
+describe('state.children', () => {
+  it('should type state children correctly', () => {
```

Comment:
All of the related functionality should kinda be removed from this PR. The PR's title is: "[v5] Input" but yet it messes with `.children`.

Of course, we want to improve the types for `children` and so on - but that should be a separate unit of work. Those PRs tend to be a little bit big and it's best to have a focused place for a discussion related to the goal of the PR.